### PR TITLE
feat: add Lenses to app-utils

### PR DESCRIPTION
**Pull request by civsiv**

### CODE REVIEW

File: src/lenses.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      return lensByOpportunityType.scs 
+        ? R.set(lensByOpportunityType.scs, newValue, opportunity)
+        : () => { throw new Error('lens not expected to be used'); };
+    }
+    return lensByOpportunityType.slot 
+      ? R.set(lensByOpportunityType.slot, newValue, opportunity)
+      : () => { throw new Error('lens not expected to be used'); };
+  };
+
+  // @ts-ignore
```
Comment (civsiv) [inline src/lenses.js:63]: If you can think of a better way to not ts-ignore, lmk

File: src/lenses.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    scs: R.lensPath(['remainingAttendeeCapacity']),
+    slot: R.lensPath(['remainingUses']),
+  }),
+  /** @type {import('ramda').Lens<TOpportunity, number> } */
+  maxCapacity: opportunityTypeLens({
+    scs: R.lensPath(['maximumAttendeeCapacity']),
+    slot: R.lensPath(['maximumUses']),
+  }),
+  /** @type {import('ramda').Lens<TOpportunity, SearchIsBookingRequestFacilityUseType>} */
+  facilityUse: opportunityTypeLens({
```
Comment (civsiv) [inline src/lenses.js:109]: Have added this (compared to the Booking lenses.js)

File: src/lenses.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  return R.lens(slotGetter, slotSetter);
+}
+
+/**
+ * @param {{
+ *  scs?: import('ramda').Lens<BsBookableScheduledSessionType, any>,
+ *  slot?: import('ramda').Lens<SearchIsBookingRequestFacilityUseSlotType, any>
+ * }} lensByOpportunityType
+ */
+function opportunityTypeLens(lensByOpportunityType) {
```
Comment (civsiv) [inline src/lenses.js:40]: Have changed this function to have null `slot` or `scs` compared to Booking lenses.js

**Comment by lukehesluke** [comment]
[Booking: Multiparty Cancellations](https://trello.com/c/CRmMIDQW/4335-booking-multiparty-cancellations)

**Pull request merged**